### PR TITLE
Allow spaces in regex

### LINKED ISSUE #94: Syntax highlighting gets thrown off by regular expressions
Hi,

I noticed that if I have a line like this in a coffeescript file:

```coffeescript
values.map((v) -> v.value.toLowerCase().replace(/ /g, '_')).join('_').replace(/[^0-9a-z_]/g, '')
```

...the remaining code in the file will not be highlighted correctly. (in fact, it appears that GitHub gets it wrong too)

I am running language-coffee-script version 0.46.1.

### PR DESCRIPTION
**Pull request by 50Wliu**
Fixes #94

/cc @esdoppio

### CODE REVIEW

File: grammars/coffeescript.cson [hunk 1]
```diff
@@ -139,7 +139,7 @@
 
   }
   {
-    'begin': '/(?=(?![ /*+?])([^\\\\]|\\\\.)*?/[igmy]{0,4}(?![a-zA-Z0-9]))'
+    'begin': '/(?=(?![/*+?])([^\\\\]|\\\\.)*?/[igmy]{0,4}(?![a-zA-Z0-9]))'
```
Comment (esdoppio) [inline grammars/coffeescript.cson]: Now that we're on this, consider further improve it to

```
/(?![/*+?])(?=([^\\\\]|\\\\.)*?/[gimuy]*)
```

and with the end pattern 😉

**Comment by 50Wliu** [comment]
These regex patterns are actually hilariously bad.  First of all they were using javascript scopes (and still are for the actual regex), and then both the begin and end slashes were being tokenized as the beginning ones.

File: grammars/coffeescript.cson [hunk 2]
```diff
@@ -139,20 +138,20 @@
 
   }
   {
-    'begin': '/(?=(?![/*+?])([^\\\\]|\\\\.)*?/[igmy]{0,4}(?![a-zA-Z0-9]))'
-    'end': '(/)[imgy]{0,4}'
+    'begin': '/(?=(?![/*+?])([^\\\\]|\\\\.)*?/[gimuy]*(?![a-zA-Z0-9]))'
```
Comment (esdoppio) [inline grammars/coffeescript.cson]: 😂  That's not exactly what I suggested, in response to the commit msg.

Comment (50Wliu) [inline grammars/coffeescript.cson]: Whoops!  I only noticed the gimuy change for some reason.

**Comment by 50Wliu** [comment]
I didn't change the block regexp to use the extended lookahead because we already know we're in a regex with that one.

**Comment by 50Wliu** [comment]
Does this look good now @esdoppio?

**Comment by esdoppio** [comment]
👍 
Sorry, I must have missed the notice.

**Pull request merged**